packages/expo-doctor/src/checks/InstalledDependencyVersionCheck.ts

```diff
@@ -0,0 +1,38 @@
+import spawnAsync, { SpawnResult } from '@expo/spawn-async';
+
+import { DoctorCheck, DoctorCheckParams, DoctorCheckResult } from './checks.types';
+
+function isSpawnResult(result: any): result is SpawnResult {
+  return 'stderr' in result && 'stdout' in result && 'status' in result;
+}
+
+export class InstalledDependencyVersionCheck implements DoctorCheck {
+  description = 'Checking dependency versions for compatibility with the installed Expo SDK';
+
+  async runAsync({ projectRoot }: DoctorCheckParams): Promise<DoctorCheckResult> {
+    const issues: string[] = [];
+    const advice: string[] = [];
+
+    try {
+      // only way to check dependencies without automatically fixing them is to use interactive prompt
+      // In the future, we should add JSON output to npx expo install, check for support, and use that instead
+      await spawnAsync('sh', ['-c', 'echo "n" | npx expo install --check'], {
+        stdio: 'pipe',
+        cwd: projectRoot,
+      });
+    } catch (error: any) {
+      if (isSpawnResult(error)) {
+        issues.push(error.stderr.trim());
```

Comment:
Add a debug-only log which prints the entire error

Reply:
Now that the outer-try catch lets `Log.exception()` do the printing, we at least get the full exception for unexpected situations. Any advice for doing the debug logging combined with the ora spinner so things don't get out of order? With the regular "issues" output, I just delay printing until the check is done. The extra info from the error here (e.g., `Error: sh exited with non-zero code: 1`) doesn't add much, it'd be good to keep it separate.